From 0d050227c0ddb79280dde493b0cada650ab2fa30 Mon Sep 17 00:00:00 2001
From: Alexandre Torgue <alexandre.torgue@st.com>
Date: Wed, 23 Sep 2020 12:14:03 +0200
Subject: [PATCH] ARM: dts: stm32: Add power domain on stm32mp151

Enable stm32-pm-domain driver and add two core power domains:
core and core_ret power domains.

Signed-off-by: Alexandre Torgue <alexandre.torgue@st.com>
Change-Id: Ifa591d7d6d914372e4bc36feee1f3bf9cf2040f1
---
 arch/arm/boot/dts/st/stm32mp151.dtsi | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

--- a/arch/arm/boot/dts/st/stm32mp151.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp151.dtsi
@@ -92,6 +92,26 @@
 		};
 	};
 
+	pm_domain {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "st,stm32mp157c-pd";
+
+		pd_core_ret: core-ret-power-domain@1 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <1>;
+			#power-domain-cells = <0>;
+			label = "CORE-RETENTION";
+
+			pd_core: core-power-domain@2 {
+				reg = <2>;
+				#power-domain-cells = <0>;
+				label = "CORE";
+			};
+		};
+	};
+
 	thermal-zones {
 		cpu_thermal: cpu-thermal {
 			polling-delay-passive = <0>;
